src/main/kotlin/com/sourcegraph/cody/chat/Chat.kt

```diff
@@ -77,30 +79,29 @@ class Chat {
         val rateLimitError = throwable.toRateLimitError()
         RateLimitStateManager.reportForChat(project, rateLimitError)
 
-        // TODO(mikolaj):
-        // RFC 872 mentions `feature flag cody-pro: true`
-        // the flag should be a factor in whether to show the upgrade option
-        val isGa = java.lang.Boolean.getBoolean("cody.isGa")
-        val text =
-            when {
-              rateLimitError.upgradeIsAvailable && isGa -> {
-                "<b>You've used up your chat and commands for the month:</b> " +
-                    "You've used all${rateLimitError.limit?.let { " $it" }} chat messages and commands for the month. " +
-                    "Upgrade to Cody Pro for unlimited autocompletes, chats, and commands. " +
-                    "<a href=\"https://sourcegraph.com/cody/subscription\">Upgrade</a> " +
-                    "or <a href=\"https://sourcegraph.com/cody/subscription\">learn more</a>.<br><br>" +
-                    "(Already upgraded to Pro? Restart your IDE for changes to take effect)"
-              }
-              else -> {
-                "<b>Request failed:</b> You've used all${rateLimitError.quotaString()} chat messages and commands." +
-                    " The allowed number of request per day is limited at the moment to ensure the service stays functional.${rateLimitError.resetString()} " +
-                    "<a href=\"https://docs.sourcegraph.com/cody/core-concepts/cody-gateway#rate-limits-and-quotas\">Learn more.</a>"
+        ApplicationManager.getApplication().executeOnPooledThread {
+          val codyProJetbrains = isCodyProJetbrains(project)
+          val text =
+              when {
+                rateLimitError.upgradeIsAvailable && codyProJetbrains -> {
+                  "<b>You've used up your chat and commands for the month:</b> " +
+                      "You've used all${rateLimitError.limit?.let { " $it" }} chat messages and commands for the month. " +
+                      "Upgrade to Cody Pro for unlimited autocompletes, chats, and commands. " +
+                      "<a href=\"https://sourcegraph.com/cody/subscription\">Upgrade</a> " +
+                      "or <a href=\"https://sourcegraph.com/cody/subscription\">learn more</a>.<br><br>" +
+                      "(Already upgraded to Pro? Restart your IDE for changes to take effect)"
+                }
+                else -> {
+                  "<b>Request failed:</b> You've used all${rateLimitError.quotaString()} chat messages and commands." +
+                      " The allowed number of request per day is limited at the moment to ensure the service stays functional.${rateLimitError.resetString()} " +
+                      "<a href=\"https://docs.sourcegraph.com/cody/core-concepts/cody-gateway#rate-limits-and-quotas\">Learn more.</a>"
+                }
```

Comment:
I've just noticed, but we can use a multiline string everywhere for readability.